archivebox/extractors/__init__.py

```diff
@@ -136,24 +128,34 @@ def archive_link(link: Link, overwrite: bool=False, methods: Optional[Iterable[s
 
     return link
 
-
 @enforce_types
-def archive_links(links: List[Link], overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
-    if not links:
+def archive_links(all_links: any, overwrite: bool=False, methods: Optional[Iterable[str]]=None, out_dir: Optional[str]=None) -> List[Link]:
+
+    if type(all_links) is list:
+        num_links: int = len(all_links)
+        get_link = lambda x: x
+        get_iter = lambda x: x
+    else:
+        num_links: int = all_links.count()
+        get_link = lambda x: x.as_link()
+        get_iter = lambda x: x.iterator()
```

Comment:
```suggestion
    if isinstance(all_links, QuerySet):
        all_links = all_links.iterator()
        num_links = all_links.count()
    else:
        num_links = len(all_links)
```

Reply:
I need to differentiate between a `Snapshot` and a `Link`. If I do this, I would have to check the type in the loop again, because if it is an snapshot, I need to run `as_link` to be able to process it (until we refactor everything to use Snapshot at least). I got rid of the `get_iter` lambda, and shadowed the `all_links` variable (not a fan of shadowing inputs tho): https://github.com/pirate/ArchiveBox/pull/452/commits/4ba3572919b180d1360c4e2f2ffa81be950b5f8a